cmd/clusterctl/pkg/client/repository/client.go

```diff
@@ -163,8 +163,14 @@ func repositoryFactory(providerConfig config.Provider, configVariablesClient con
 		return repo, err
 	}
 
-	// if the url is a local repository
-	//TODO: implement in a follow up PR
+	// if the url is a local filesystem repository
+	if rURL.Scheme == "file" {
```

Comment:
Can we add some tests for this?

Reply:
Unfortunately, I don't see a simple manner by which to exercise this case in a unit test. The closest thing I see is the `TestNewFakeClient` but that does not appear to support a flexible set of urls.
 https://github.com/kubernetes-sigs/cluster-api/blob/065eb539766dede097e206a7b549b5902d15f14a/cmd/clusterctl/pkg/client/client_test.go#L34
There is also this but that does not exercise the above code path:
https://github.com/kubernetes-sigs/cluster-api/blob/065eb539766dede097e206a7b549b5902d15f14a/cmd/clusterctl/pkg/client/config/providers_client_test.go#L28

I'm currently working on a e2e framework. Maybe we can test this code path there?

Reply:
Would it be possible to create a temp dir, populate it with some repo files, and then create a real repository client and make sure that "file" and "" schemes resolve to the local repo?

Reply:
Done PTAL